doc/scss/custom.scss

```diff
@@ -190,3 +190,65 @@ div.sk-text-image-grid-small {
 div.sk-text-image-grid-large {
   @include sk-text-image-grid(100px);
 }
+
+/* Style of the docsearch modal window */
+
+// In the following, we mainly reuse the background color and font-color from the
+// pydata-sphinx-theme. We also adjust slightly the font-size to be wider and in line
+// with the sidebar design.
+:root {
+  --docsearch-modal-width: 768px;
+  --docsearch-modal-height: 80vh;
+  --docsearch-searchbox-shadow: inset 0 0 0 2px var(--pst-color-primary);
+  --docsearch-highlight-color: var(--pst-color-primary);
+}
+
+.DocSearch-Button-Key {
+  top: 0;
+}
+
+@media screen and (max-width: 1200px) {
+  .DocSearch-Button-Placeholder,
+  .DocSearch-Button-Keys {
+    display: none;
+  }
+}
+
+.DocSearch-Modal {
+  background: var(--pst-color-background);
+  margin-top: 10vh;
+}
+
+.DocSearch-Form {
+  background: var(--pst-color-background);
+}
+
+.DocSearch-MagnifierLabel,
+.DocSearch-Reset {
+  color: var(--pst-color-primary);
+}
+
+.DocSearch-Input {
+  color: var(--pst-color-text-base);
+  font-size: var(--pst-font-size-icon);
+}
+
+.DocSearch-Hit-Icon svg {
```

Comment:
Uhm weird, it seems that we still inherit from the `bootstrap.css` style here. @Charlie-XIAO do you see my mistake here?

Reply:
`pydata-sphinx-theme` is itself bootstrap-based, and it also makes customizations on top of bootstrap.

Reply:
Oh what I meant is the following:

<img width="333" alt="image" src="https://github.com/scikit-learn/scikit-learn/assets/7454015/5038d674-14f5-48dd-84c7-fa2ed9d058b7">

We are not using the CSS rule that we intend to.

Reply:
Sorry, stupid typo `Icon` -> `icon` :)

Reply:
My bad 🤣